tests/test_wire/test_rewire.py

```diff
@@ -11,3 +11,27 @@ def test_anonymous_bits_rewiring_warning(first, second, caplog):
     x = m.as_bits(x)
     x @= second
     assert has_warning(caplog)
+
+
+@pytest.fixture
+def _setup_and_teardown():
+    prev = m.config.config.rewire_log_level
+    m.config.config.rewire_log_level = "ERROR"
+    yield
+    m.config.config.rewire_log_level = prev
+
+
+def test_rewire_error(caplog, _setup_and_teardown):
+    x = m.Bit(name="x")
+    x @= 0
+    x @= 1
+    assert has_error(caplog)
+    # Check that it makes the circuit def have an error
+
+    class Foo(m.Circuit):
+        io = m.IO(O=m.Out(m.Bit))
+        io.O @= 0
+        io.O @= 1
```

Comment:
```suggestion

    # NOTE(leonardt): We additionally check that compiling a circuit with a                                                              
    # rewiring error causes an exception (just a double-check since this                                                                 
    # functionality is independent of the type of error logged).                                                                         
    class Foo(m.Circuit):
        io = m.IO(O=m.Out(m.Bit))
        io.O @= 0
        io.O @= 1
```